### PR TITLE
API: Added pull review read only endpoints

### PR DESCRIPTION
**Pull request by tbe**
API Endpoints for Reviews.

implements #5733

### CODE REVIEW

File: modules/structs/pull_review.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// PullRequestReview represents a pull request review
+type PullRequestReview struct {
+	ID       int64  `json:"id"`
+	PRURL    string `json:"pull_request_url"`
+	Reviewer *User  `json:"user"`
+	Body     string `json:"body"`
+	CommitID string `json:"commit_id"`
+	Type     string `json:"type"`
+
```
Comment (6543) [inline modules/structs/pull_review.go:19]: can you add state info too?
```json
"state": "APPROVED",
```

Comment (6543) [reply modules/structs/pull_review.go:19]: o it still a draft :D - the start looks good  :+1:

Comment (tbe) [reply modules/structs/pull_review.go:19]: I thought about naming that `state`, but it was so misleading, that I've gone with `type`.
```
"state":"COMMENT"
```
just looks strange.

But i can rename this if wanted.

Comment (6543) [reply modules/structs/pull_review.go:19]: fair point ... just did a comparsion with the github api ...
and beside COMMENT, PENDING, APPROVED, REJECT could be interpreted as state ...

Comment (6543) [reply modules/structs/pull_review.go:19]: > But i can rename this if wanted.

please :)

**Comment by bagasme** [comment]
@tbe GH API reference?

**Comment by 6543** [comment]
@bagasme this PR is still a DRAFT and the link to github api docu is this: https://developer.github.com/v3/pulls/reviews

File: modules/structs/pull_review.go (hunk 1 above)
Comment (davidsvantesson) [reply modules/structs/pull_review.go:19]: Maybe add field `submitted_at`?

Comment (davidsvantesson) [reply modules/structs/pull_review.go:19]: `stale` and `official` can also be interesting fields (GH doesn't have them).
Also possibility to get list of only latest review of each reviewer would be good, as that is what is seen on PR page.

**Comment by 6543** [comment]
@tbe what's the status?

**Comment by 6543** [comment]
@tbe am I allowed to take over this?

**Comment by 6543** [comment]
ok I'll take this

**Comment by 6543** [comment]
work will go on in #11224

**Comment by 6543** [comment]
@tbe I have finished it :)

If you like you can have a look at #11224

**Pull request closed without merging**